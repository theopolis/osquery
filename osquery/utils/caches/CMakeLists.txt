# Copyright (c) 2014-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed in accordance with the terms specified in
# the LICENSE file found in the root directory of this source tree.

function(osqueryUtilsCachesMain)

  if(OSQUERY_BUILD_TESTS)
    add_subdirectory("tests")
  endif()

  generateOsqueryUtilsCachesLru()
endfunction()

function(generateOsqueryUtilsCachesLru)
  add_library(osquery_utils_caches_lru INTERFACE)

  set(public_header_files
    lru.h
    lru-impl.h
  )

  generateIncludeNamespace(osquery_utils_caches_lru "osquery/utils/caches" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_caches_tests_lrutests-test COMMAND osquery_utils_caches_tests_lrutests-test)
endfunction()

osqueryUtilsCachesMain()
