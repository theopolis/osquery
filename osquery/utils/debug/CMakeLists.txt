# Copyright (c) 2014-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed in accordance with the terms specified in
# the LICENSE file found in the root directory of this source tree.

function(osqueryUtilsDebugMain)

  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsDebugDebugtestsTest()
  endif()

  generateOsqueryUtilsDebug()
endfunction()

function(generateOsqueryUtilsDebug)
  add_osquery_library(osquery_utils_debug INTERFACE)

  target_link_libraries(osquery_utils_debug INTERFACE
    global_cxx_settings
    thirdparty_boost
  )

  set(public_header_files
    debug_only.h
  )

  generateIncludeNamespace(osquery_utils_debug "osquery/utils/debug" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_debug_debugtests-test COMMAND osquery_utils_debug_debugtests-test)

endfunction()

function(generateOsqueryUtilsDebugDebugtestsTest)
  add_osquery_executable(osquery_utils_debug_debugtests-test tests/debug_only.cpp)

  target_link_libraries(osquery_utils_debug_debugtests-test PRIVATE
    global_cxx_settings
    osquery_utils_debug
    thirdparty_googletest
  )
endfunction()

osqueryUtilsDebugMain()
