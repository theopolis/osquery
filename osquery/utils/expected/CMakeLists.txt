# Copyright (c) 2014-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed in accordance with the terms specified in
# the LICENSE file found in the root directory of this source tree.

function(osqueryUtilsExpectedMain)
  if(OSQUERY_BUILD_TESTS)
    generateOsqueryUtilsExpectedExpectedtestsTest()
  endif()

  generateOsqueryUtilsExpected()
endfunction()

function(generateOsqueryUtilsExpected)
  add_osquery_library(osquery_utils_expected INTERFACE)

  target_link_libraries(osquery_utils_expected INTERFACE
    global_cxx_settings
    osquery_utils_debug
    osquery_utils_error
    thirdparty_boost
  )

  set(public_header_files
    expected.h
  )

  generateIncludeNamespace(osquery_utils_expected "osquery/utils/expected" "FILE_ONLY" ${public_header_files})

  add_test(NAME osquery_utils_expected_expectedtests-test COMMAND osquery_utils_expected_expectedtests-test)

endfunction()

function(generateOsqueryUtilsExpectedExpectedtestsTest)
  add_osquery_executable(osquery_utils_expected_expectedtests-test tests/expected.cpp)

  target_link_libraries(osquery_utils_expected_expectedtests-test PRIVATE
    global_cxx_settings
    osquery_utils_expected
    thirdparty_googletest
  )
endfunction()

osqueryUtilsExpectedMain()
